core/dbt/deprecations.py

```diff
@@ -176,6 +184,7 @@ def warn(name, *args, **kwargs):
     ModelsKeyNonModelDeprecation(),
     ExecuteMacrosReleaseDeprecation(),
     AdapterMacroDeprecation(),
+    PackageRedirectDeprecation()
```

Comment:
As far as testing this, we can take a cue from the existing [deprecation tests](https://github.com/dbt-labs/dbt/blob/develop/test/integration/012_deprecation_tests/test_deprecations.py):
- set up the conditions for the deprecation to raise
- assert that it raises as a warning
- assert that the warning becomes an error in [`--warn-error`](https://docs.getdbt.com/reference/global-cli-flags#warnings-as-errors) mode (called `strict` here for historical reasons)
- that the warning/exception message says what you think it should say

So, you can add this or something like it to the bottom of `test_deprecations.py`. The methods prefixed with `test_postgres_` will be picked up during Postgres integration tests.
```python
class TestPackageRedirectDeprecation(BaseTestDeprecations):
    @property
    def models(self):
        return self.dir('where-were-going-we-dont-need-models')

    @property
    def packages_config(self):
        return {
            "packages": [
                {
                    'package': 'fishtown-analytics/dbt_utils',
                    'version': '0.7.0'
                }
            ]
        }
    
    @use_profile('postgres')
    def test_postgres_package_redirect(self):
        self.assertEqual(deprecations.active_deprecations, set())
        self.run_dbt(['deps'], strict=False)
        expected = {'package-redirect'}
        self.assertEqual(expected, deprecations.active_deprecations)

    @use_profile('postgres')
    def test_postgres_package_redirect_fail(self):
        self.assertEqual(deprecations.active_deprecations, set())
        with self.assertRaises(dbt.exceptions.CompilationException) as exc:
            self.run_dbt(['deps'], strict=True)
        exc_str = ' '.join(str(exc.exception).split())  # flatten all whitespace
        expected = "The `fishtown-analytics/dbt_utils` package is deprecated in favor of `dbt-labs/dbt_utils`"
        assert expected in exc_str
```